### ISSUE TITLE
Can't set field in mocked struct-argument

### ISSUE THREAD

**Issue by thoni56** [cgreen-devs/cgreen#216]
Having  the following

```
typedef struct {
    int field;
} Struct;

void f(Struct *sp) {
    mock(sp);
}
```

How can I set the value of `field` in the struct passed in by reference?

`with_sideeffect()` can't reference mocked arguments at all, and `will_set_contents_of()` cannot get at the field.

**Comment by souryogurt** [comment]
I doing something like this usually:
```c
typedef struct {
    int field;
} Struct;

void f(Struct *sp) {
    int *field = &sp->field;
    mock(sp, field);
}

Ensure(mocked_function_sets_field)
{
    Struct some = {0};
    int new_value = 5;
    expect(f, will_set_contents_of_parameter(field, &new_value, sizeof(new_value));
    assert_that(some.field, is_equal_to(new_value));
}
```

**Comment by thoni56** [comment]
Woah! I didn't actually expect someone to be monitoring this, much less to get some help from a like minded person ;-). Thanks!

Yeah, that kind of works since `mock` doesn't care what arguments it gets, so you can "mock" anything. Drawback is that you need to go into your mocked functions to fix it, but I suppose I can live with that.

Giving the temporary "mock" arguments names like (in your example) `sp_field` will even be somewhat readable.

Thanks, that solves my immediate issue, maybe that should even go in the tutorial, since at this time there is no better way to handle it.

I'll keep this issue around for the future. I was thinking along the lines of 

    with_sideeffect_using_argument(func, sp)

**Comment by thoni56** [comment]
Added this to the 1.4.0 milestone, but that is aimed at the documentation about the struct fields mentioned above.

**Comment by thoni56** [comment]
Closed by adding a section 3.3.2 of the guide.
